dataprofiler/data_readers/graph_data.py

```diff
@@ -169,12 +169,12 @@ def is_match(cls, file_path, options=None):
             return True
         return False
 
-    def _format_data_networkx(self):
+    def _format_data_networkx(self) -> nx.Graph:
         """Format the input file into a networkX graph."""
         networkx_graph = nx.Graph()
 
         # read lines from csv
-        if not self._checked_header or not self._delimiter:
+        if (not self._checked_header or not self._delimiter) and self.input_file_path is not None:
```

Comment:
why does this need to be not `None`? was it not working before?

Does `load_as_str_from_file` need to update to accept `Optional` as it handles it internally?

Reply:
`input_file_path` cannot be None, fair. Is there a better way to define at the beginning of the func as opposed to repeating throughout the func?

Could we at the beginning of the func do:
```python
assert self.input_file_path is not None
```